common-java-libs/org.wso2.apk.apimgt.impl/src/main/java/org/wso2/apk/apimgt/impl/APIManagerFactory.java

```diff
@@ -108,10 +108,31 @@ public APIConsumer getAPIConsumerFromCache(String key, ConsumerCreator consumerC
         return consumer;
     }
 
+    public APIProvider getAPIProviderFromCache(String key, ProviderCreator providerCreator)
+            throws APIManagementException {
+        APIProvider provider = providers.get(key);
```

Comment:
lets remove this cache

Reply:
addressed with f0cddb250859adb268acb465a0f427f1da0e5766